pkg/runtime.v2/framework/core/framework.go

```diff
@@ -104,10 +104,10 @@ func (f *Framework) RunCustomValidationPlugins(oldObj, newObj *kubeflowv2.TrainJ
 	return aggregatedWarnings, aggregatedErrors
 }
 
-func (f *Framework) RunComponentBuilderPlugins(ctx context.Context, info *runtime.Info, trainJob *kubeflowv2.TrainJob) ([]client.Object, error) {
+func (f *Framework) RunComponentBuilderPlugins(ctx context.Context, runtimeJobTemplate client.Object, info *runtime.Info, trainJob *kubeflowv2.TrainJob) ([]client.Object, error) {
 	var objs []client.Object
 	for _, plugin := range f.componentBuilderPlugins {
-		obj, err := plugin.Build(ctx, info, trainJob)
+		obj, err := plugin.Build(ctx, runtimeJobTemplate, info, trainJob)
```

Comment:
ctx context is not getting used anywhere throughout the code, is it necessary to pass?

Reply:
I think, we use it here to get the existing object and compare it
https://github.com/kubeflow/training-operator/blob/master/pkg/runtime.v2/framework/plugins/jobset/jobset.go#L82-L84